### PR TITLE
ENT4441-Send LMS User ID with bulk code

### PR DESCRIPTION
**Pull request by zamanafzal**

### CODE REVIEW

File: src/components/CodeAssignmentModal/index.jsx [hunk 1]
```diff
@@ -97,6 +99,19 @@ export class BaseCodeAssignmentModal extends React.Component {
     this.setState({ mode });
   }
 
+  getCleanedUsersDetails(emails, usersDetailResponse) {
```
Comment (HammadAhmadWaqas) [inline src/components/CodeAssignmentModal/index.jsx:102]: can we rename from `getCleanedUsersDetails ` to `getCleanedUserDetails`
and 
`usersDetailResponse` to `userDetailsResponse`

File: src/components/CodeAssignmentModal/index.jsx [hunk 2]
```diff
@@ -97,6 +99,19 @@ export class BaseCodeAssignmentModal extends React.Component {
...
+    const usersDetail = [];
```
Comment (HammadAhmadWaqas) [inline src/components/CodeAssignmentModal/index.jsx:103]: can we rename it to `userDetails`

File: src/components/CodeAssignmentModal/index.jsx [hunk 3]
```diff
@@ -285,6 +300,15 @@ export class BaseCodeAssignmentModal extends React.Component {
       };
     }
 
+    let usersDetailResponse = [];
```
Comment (HammadAhmadWaqas) [inline src/components/CodeAssignmentModal/index.jsx:303]: can we rename it to `usersDetailResponse` to `userDetailsResponse `

File: src/components/CodeAssignmentModal/index.jsx [hunk 4]
```diff
@@ -285,6 +300,15 @@ export class BaseCodeAssignmentModal extends React.Component {
...
+    try {
+      const response = await LmsApiService.fetchUserDetailsFromEmail(options.emails);
+      usersDetailResponse = response.data;
+    } catch (error) {
+      logError(error);
+    }
+    options.users_details = this.getCleanedUsersDetails(options.emails, usersDetailResponse);
```
Comment (HammadAhmadWaqas) [inline src/components/CodeAssignmentModal/index.jsx:310]: We need to rename it to `options.user_details`

File: src/components/CodeAssignmentModal/validation.js [hunk 5]
```diff
@@ -25,6 +27,7 @@ export const getInvalidEmailMessage = (invalidEmailIndices, emails) => {
 
 export const NO_EMAIL_ADDRESS_ERROR = 'No email addresses provided. Either manually enter email addresses or upload a CSV file.';
 export const BOTH_TEXT_AREA_AND_CSV_ERROR = 'You uploaded a CSV and manually entered email addresses. Please only use one of these fields.';
+export const MAX_EMAILS_ADDRESS_ALLOWED_ERROR = 'You can add maximum of 500 Email Addresses at a time.';
```
Comment (HammadAhmadWaqas) [inline src/components/CodeAssignmentModal/validation.js:30]: can we update the message?
```
You have uploaded more than 500 email addresses, Please upload 500 or less.
```

**Review by HammadAhmadWaqas — COMMENTED** [review]
Looks good to me, just a few variable rename requests.

File: src/components/CodeAssignmentModal/index.jsx (hunk 4 above)
Comment (zamanafzal) [reply src/components/CodeAssignmentModal/index.jsx:310]: I think the user should be plural here as we are having multiple user data in this array.

File: src/components/CodeAssignmentModal/index.jsx [hunk 6]
```diff
@@ -285,6 +300,16 @@ export class BaseCodeAssignmentModal extends React.Component {
...
 
+    let usersResponse = [];
+    try {
+      const response = await LmsApiService.fetchUserDetailsFromEmail({ emails: options.emails });
+      usersResponse = response.data;
+    } catch (error) {
+      logError(error);
+      throw new SubmissionError({ _error: error });
+    }
+    options.users = this.getCleanedUsers(options.emails, usersResponse);
```
Comment (zamanafzal) [inline src/components/CodeAssignmentModal/index.jsx:311]: @HammadAhmadWaqas Please verify that you are expecting values in `users` variable.

Comment (HammadAhmadWaqas) [reply src/components/CodeAssignmentModal/index.jsx:311]: yes looks good.

**Review by sameenfatima78 — APPROVED** [review]
LGTM!

File: src/components/CodeAssignmentModal/index.jsx [hunk 7]
```diff
@@ -97,6 +99,19 @@ export class BaseCodeAssignmentModal extends React.Component {
     this.setState({ mode });
   }
 
+  getCleanedUsers(emails, usersResponse) {
+    const users = [];
+    emails.forEach((email) => {
+      const user = usersResponse.find(userResponse => userResponse.email === email);
+      users.push({
+        email,
+        lms_user_id: user ? user.id : undefined,
```
Comment (HammadAhmadWaqas) [inline src/components/CodeAssignmentModal/index.jsx:108]: we don't want to send these keys if the value is not there.

Comment (zamanafzal) [reply src/components/CodeAssignmentModal/index.jsx:108]: Key will not be sent if it is undefined.

**Pull request merged**